Sources/Private/Experimental/ValueProviderStore.swift

```diff
@@ -12,8 +14,8 @@ final class ValueProviderStore {
   /// Registers an `AnyValueProvider` for the given `AnimationKeypath`
   func setValueProvider(_ valueProvider: AnyValueProvider, keypath: AnimationKeypath) {
     LottieLogger.shared.assert(
-      valueProvider.typeErasedStorage.isSingleValue,
-      "The new rendering engine only supports Value Providers with a single fixed value")
+      valueProvider.typeErasedStorage.isSupportedByCoreAnimationRenderingEngine,
+      "The Core Animation rendering engine doesn't support Value Providers that vend a closure.")
```

Comment:
nit: might make sense to add a comment detailing that it is impossible to have an API like this with Core Animation, as-is this sounds like it may be added in the future